### PR TITLE
Implementing user form to signup a profile

### LINKED ISSUE #54: User profile information form
When a new user comes to our website, and logins via Github, we redirect them to /goto which then decides where the user was before the login process began, and should take them back there. Currently, /goto always drops the user back to the www site

Upon login via Github for a new user, we create a temporary id for them, we want to change this into an RDS username with certain rules
- Starts with users's first name
- No special characters, except hyphens
- No funky unprofessional words in the username
- Avoid numbers

We want to create a page with a form where a new user can select a username, and fill out or correct their first name, last name, YoE, Company, etc (We can pull in the initial name from Github during login)

We need to create this page to allow new users to sign up to our platform

### PR DESCRIPTION
**Pull request by ankushdharkar**
This allows the user to create profile for the very first time

Closes #54

### CODE REVIEW

File: js/user-form.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    DESIGNATION,
+    GITHUB_ID,
+    LINKEDIN_ID,
+    TWITTER_ID
+  ]);
+
+  console.log(dataToSend);
+
+  try {
+    const res = await fetch(`AWAITING_API__https://staging-api.realdevsquad.com/users/self`, {
```
Comment (ankushdharkar) [inline js/user-form.js:143]: @Rucha1499 I am ready to start using the API. Let me know when it's ready please

File: js/user-form.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    TWITTER_ID
+  ]);
+
+  updateUserData();
+});
+
+function getDataToSend (arrKeys) {
+  return arrKeys.reduce(function(acc, k) {
+    const el = document.getElementById(k);
+
```
Comment (sumitd94) [inline js/user-form.js:116]: I think we can be more specific with the arguments names being passed to the anonymous function on the `reduce`
Like if I directly jump on this function, I have no clue what `acc` and `k` means

File: js/user-form.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    EMAIL,
+    PHONE, 
+    YOE,
+    COMPANY,
+    DESIGNATION,
+    GITHUB_ID,
+    LINKEDIN_ID,
+    TWITTER_ID
+  ]);
+
```
Comment (sumitd94) [inline js/user-form.js:139]: Lets create a constant at the top and store this argument which is being passed to `getDataToSend` in it and then use that constant as argument.
similarly for the code on line no 98

File: js/user-form.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  console.log(dataToSend);
```
Comment (swarajpure) [inline js/user-form.js:140]: Are we keeping this console log statement?

Comment (ankushdharkar) [reply js/user-form.js:140]: No. I was just waiting for the API to start working. Will remove this in the final version. Thanks!

File: js/user-form.js (hunk 2 above)
Comment (ankushdharkar) [reply js/user-form.js:116]: Great point @sumitd94 

Array.prototype.reduce has a pretty standard sequence according to the positions.
acc - accumulator
k - key

Comment (sumitd94) [reply js/user-form.js:117]: Yep

**Pull request merged**